cmd/tink-controller-v1alpha2/main.go

```diff
@@ -0,0 +1,179 @@
+	Kubeconfig           string // only applies to out of cluster
+	MetricsAddr          string
+	ProbeAddr            string
+	EnableLeaderElection bool
+}
+
+func (c *Config) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&c.K8sAPI, "kubernetes", "",
+		"The Kubernetes API URL, used for in-cluster client construction.")
+	fs.StringVar(&c.Kubeconfig, "kubeconfig", "", "Absolute path to the kubeconfig file")
+	fs.StringVar(&c.MetricsAddr, "metrics-bind-address", ":8080",
+		"The address the metric endpoint binds to.")
+	fs.StringVar(&c.ProbeAddr, "health-probe-bind-address", ":8081",
+		"The address the probe endpoint binds to.")
+	fs.BoolVar(&c.EnableLeaderElection, "leader-elect", false,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.")
+}
+
+func main() {
+	cmd := NewRootCommand()
+	if err := cmd.Execute(); err != nil {
+		os.Exit(1)
+	}
+}
+
+func NewRootCommand() *cobra.Command {
+	var config Config
+
+	zlog, err := zap.NewProduction()
+	if err != nil {
+		panic(err)
+	}
+	logger := zapr.NewLogger(zlog).WithName("github.com/tinkerbell/tink")
+
+	cmd := &cobra.Command{
+		Use: "tink-controller",
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			viper, err := createViper(logger)
+			if err != nil {
+				return fmt.Errorf("config init: %w", err)
+			}
+			return applyViper(viper, cmd)
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			logger.Info("Starting controller version " + version)
+
+			ccfg := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+				&clientcmd.ClientConfigLoadingRules{ExplicitPath: config.Kubeconfig},
+				&clientcmd.ConfigOverrides{ClusterInfo: clientcmdapi.Cluster{Server: config.K8sAPI}})
+
+			cfg, err := ccfg.ClientConfig()
+			if err != nil {
+				return err
+			}
+
+			namespace, _, err := ccfg.Namespace()
+			if err != nil {
+				return err
+			}
+
+			opts := ctrl.Options{
+				Logger:                  logger,
+				LeaderElection:          config.EnableLeaderElection,
+				LeaderElectionID:        "tink.tinkerbell.org",
+				LeaderElectionNamespace: namespace,
+				Metrics: server.Options{
+					BindAddress: config.MetricsAddr,
+				},
+				HealthProbeBindAddress: config.ProbeAddr,
+				Scheme:                 scheme,
+			}
+
+			mgr, err := ctrl.NewManager(cfg, opts)
+			if err != nil {
+				return err
+			}
+
+			if err := mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
+				return err
+			}
+
+			if err := mgr.AddReadyzCheck("readyz", healthz.Ping); err != nil {
+				return err
+			}
+
+			if err := workflow.NewReconciler(mgr.GetClient()).SetupWithManager(mgr); err != nil {
+				return err
+			}
+
+			return mgr.Start(cmd.Context())
+		},
+	}
+	config.AddFlags(cmd.Flags())
+	return cmd
+}
+
+func createViper(logger logr.Logger) (*viper.Viper, error) {
+	v := viper.New()
+	v.AutomaticEnv()
+	v.SetConfigName("tink-controller")
+	v.AddConfigPath("/etc/tinkerbell")
```

Comment:
do we need a default location for a config file? does this provide much value, you think?

Reply:
Probably not, copy-n-paste from the existing tink-controller. If we maintain the temporary v1alpha2 binary, I'll reduce this down to bare minimum so we don't end up with cruft down the road.